core/java/iesi-core/src/main/java/io/metadew/iesi/component/http/HttpComponentService.java

```diff
@@ -72,11 +80,31 @@ public String getUri(HttpComponent httpComponent) {
                 httpComponent.getEndpoint();
     }
 
+    public HttpComponentHeader convertHeaders(HttpHeader httpHeader, UUID id) {
+        UUID uuid = UUID.randomUUID();
+        return new HttpComponentHeader(
+                uuid,
+                new HttpComponentHeaderKey(id),
+                httpHeader.getName(),
+                httpHeader.getValue()
+        );
+    }
+
+    public HttpComponentQuery convertQueries(HttpQueryParameter httpQueryParameter, UUID id) {
+        UUID uuid = UUID.randomUUID();
+        return new HttpComponentQuery(
+                uuid,
+                new HttpComponentQueryKey(id),
+                httpQueryParameter.getName(),
+                httpQueryParameter.getValue()
+        );
+    }
```

Comment:
could this be exported to a dedicated class 'HttpQueryParameterTraceService' with method 'public HttpQueryParameterTrace convert(HttpQueryParameter httpQueryParameter, ActionExecution actionExecution, String actionParameterName);

Reply:
could this be exported to a dedicated class 'HttpHeaderTraceService' with method 'public HttpHeaderTrace convert(HttpHeader httpHeader, ActionExecution actionExecution, String actionParameterName);